src/main/java/net/minecraftforge/common/crafting/FurnaceHelper.java

```diff
@@ -0,0 +1,45 @@
+package net.minecraftforge.common.crafting;
+
+import java.util.Map;
+
+import com.google.common.collect.Maps;
+
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.tileentity.TileEntityFurnace;
+
+public class FurnaceHelper 
+{
+    private static Map<Item, Integer> burnTimes, vanillaBurnTimes;
+    
+    public static Map<Item, Integer> getBurnTimes()
+    {
+        return burnTimes != null ? burnTimes : (burnTimes = Maps.newLinkedHashMap());
+    }
+    
+    public static Map<Item, Integer> getVanillaBurnTimes()
+    {
+        return vanillaBurnTimes != null ? vanillaBurnTimes : (vanillaBurnTimes = TileEntityFurnace.getBurnTimes());
+    }
```

Comment:
the whole assignment expression thing is gross to me `return (field = expr())`

either way, just make the maps static final nonnull and clear/repopulate them on invalidate. no need for them to be nullable.

Reply:
null needs less memory then an empty map, thats was my thoughts when making this...